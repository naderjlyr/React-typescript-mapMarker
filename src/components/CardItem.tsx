import React, { memo, useState } from "react";
import { TJobData } from "../types/data-models";
import {
  Heading,
  Button,
  IconTextkernel,
  StepperButton,
} from "@textkernel/oneui";
interface Props {
  job: TJobData;
  onSelect: (job: TJobData) => void;
  onRemove: (job: TJobData) => void;
}

const CardItem = ({ job, onSelect, onRemove }: Props) => {
  return (
    <div className="cm-card-item">
      <div className="left-content">
        <IconTextkernel context="primary" size={48} title="Textkernel" />
        <div className="company-name">
          <Heading align="left" context="default" level="h4">
            {job.organization_name}
          </Heading>
        </div>
        <div className="job-name">
          <Heading align="left" context="default" level="h4">
            {job.job_title}
          </Heading>
        </div>
      </div>
      <div className="bottom">
        <Button
          context="primary"
          disabled={false}
          isBlock={false}
          isInline={false}
          onClick={() => onSelect(job)}
          size="normal"
          type="submit"
        >
          Locate the Job
        </Button>
        <Button
          context="bad"
          disabled={false}
          isBlock={false}
          isInline={false}
          onClick={() => onRemove(job)}
          size="normal"
          type="submit"
        >
          Delete
        </Button>
      </div>
    </div>
  );
};

export default CardItem;
export const MemoizedCard = memo(CardItem, (prevProps, nextProps) => {
  return Object.is(prevProps, nextProps);
});
