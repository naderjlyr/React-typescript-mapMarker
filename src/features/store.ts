import { configureStore } from "@reduxjs/toolkit";
import { useDispatch } from "react-redux";
import jobSlice from "./slices/jobSlice";

const store = configureStore({
  reducer: {
    jobs: jobSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export const selectJobs = (state: RootState) => state.jobs.jobs;
export const selectTargetJob = (state: RootState) =>
  state.jobs.targetedLocation;
export const selectSearchResult = (state: RootState) => state.jobs.searchResult;

export default store;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
