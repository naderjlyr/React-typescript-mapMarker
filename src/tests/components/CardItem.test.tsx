import React from "react";
import { render, screen, RenderResult } from "@testing-library/react";
import CardItem from "../../components/CardItem";
import { JobData } from "../../models/job.model";
import { Provider } from "react-redux";
import store from "../../features/store";
const dummyData: JobData = {
  id: 123,
  job_title: "an example job title",
  organization_name: "an example organization name",
  location_coordinates: ["18.858093", "2.4294694"],
};
const renderCardItem = (): RenderResult =>
  render(
    <Provider store={store}>
      <CardItem job={dummyData} />
    </Provider>
  );
describe("Card Item Component", () => {
  test("should render job data correctly", () => {
    const { asFragment } = renderCardItem();
    expect(asFragment()).toMatchSnapshot();

    //Checking for company name and job title present in the snapshot provided.
    expect(screen.getByText(dummyData.job_title)).toBeInTheDocument();
    expect(screen.getByText(dummyData.organization_name)).toBeInTheDocument();

    //Checking for two buttons for the card appearing in the snapshot
    expect(screen.getByText(/Locate the Job/i)).toBeInTheDocument();
    expect(screen.getByText(/Delete/i)).toBeInTheDocument();
  });
});
