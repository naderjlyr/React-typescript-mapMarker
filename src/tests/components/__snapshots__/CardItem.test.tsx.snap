// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item Component should render job data correctly 1`] = `
<DocumentFragment>
  <div
    class="cm-card-item"
  >
    <div
      class="left-content"
    >
      <div
        class="IconBase--3tiBSsND2A IconBase--context_primary--2U2j03XMQL"
      >
        <svg
          aria-labelledby="title"
          class="IconBase__svg--GubJhAYC9j"
          role="img"
          style="width: 48px; height: 48px;"
          viewBox="0 0 369.7 800"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Textkernel
          </title>
          <path
            d="M334.3,800C203.4,766.9,89.9,681.8,89.9,543.8V248.3H0v-104H89.9V0H223.1V144.2H349.2v104H223.1V554c0,76.5,56.7,109.6,146.6,130.8Z"
          />
        </svg>
      </div>
      <div
        class="company-name"
      >
        <h4
          class="Heading--6KJ2J_DIuG Heading--level_h4--2hAygTxBFc"
        >
          an example organization name
        </h4>
      </div>
      <div
        class="job-name"
      >
        <h4
          class="Heading--6KJ2J_DIuG Heading--level_h4--2hAygTxBFc"
        >
          an example job title
        </h4>
      </div>
    </div>
    <div
      class="bottom"
    >
      <button
        class="Button--15fub36KGi Button--context_primary--RCe9SRC0ZW"
        type="submit"
      >
        Locate the Job
      </button>
      <button
        class="Button--15fub36KGi Button--context_bad--26Pck_OeL0"
        type="submit"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
